docs/manual/generics.tex

```diff
@@ -853,6 +856,95 @@
   void m(@Nullable Object obj)
 \end{Verbatim}
 
+\sectionAndLabel{Classes with implicit qualifier parameters}{class-qualifier-polymorphism}
+
+The Checker Framework supports type \emph{qualifier} polymorphism for classes.  This is similar to
+Java's generic classes, but is used in situations where you cannot use Java generics.
+
+To add a qualifier parameter to a class, annotate its declaration with \refqualclass{framework/qual}{HasQualifierParameter}.
+
+\begin{Verbatim}
+@HasQualifierParameter(Tainted.class)
+class StringBuffer { ... }
+\end{Verbatim}
+
+\<@HasQualifierParameter> analogizes \<StringBuffer> to a
+generic collection.  A qualifier on a use of \<StringBuffer> is treated as
+appearing both on the \<StringBuffer> and on its conceptual type argument.  That
+is:\\
+\begin{tt}
+  @Tainted StringBuffer $\approx$ @Tainted Collection<@Tainted Character>\\
+  @Untainted StringBuffer $\approx$ @Untainted Collection<@Untainted Character>
+\end{tt}
+
+If two types have different qualifier arguments, they have no subtyping
+relationship.  (This is ``invariant subtyping'', also used by Java for
+generic classes.)  In particular, \<@Untainted StringBuffer> is not a
+subtype of \<@Tainted StringBuffer>; an attepmt to cast between them (in
+either direction) will yield an \<invariant.cast.unsafe> error.
+
+If a class extends a \<@HasQualifierParameter> class (or implements a
+\<@HasQualifierParameter> interface), then that class must also be marked
+\<@HasQualifierParameter>.
+
+\subsectionAndLabel{Using class qualifier parameters in method signatures}{class-qual-param-method}
+
+In a class with a class qualifier parameter, a formal parameter type should
+sometimes have the same qualifier as the class qualifier parameter.  To express
+this, annotate both the receiver type and the parameter type with the
+polymorphic qualifier for the type system. For example,
+\begin{Verbatim}
+@HasQualifierParameter(Tainted.class)
+class Buffer {
+  void append(@PolyTainted Buffer this, @PolyTainted String s) { ... }
+}
+\end{Verbatim}
+Because \<@PolyTainted> applies to a type with a qualifier parameter, then all
+other uses of \<@PolyTainted> are instantiated to the instantiation of
```

Comment:
This is unclear.  I think you mean they depend on the actual qualifier argument (which is implicit) on the receiver.